Automattic/jetpack
# My Jetpack: support features as recommendations

## Proposed changes:

* This PR adds the skeleton for three features that will be available as modules in the recommendations on My Jetpack. All classes follow the same structure and we added a new `get_feature_status` method to the base class that will be used to override the return of `get_status`. This decision was made to simplify how we use the same class for different types of products/modules.
NOTE: the changes in this PR do not affect the current behavior/functionality. This PR was created mostly to simplify reviews.

### Other information:


## Jetpack product discussion


pbNhbs-bvL-p2

## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Spin up a JN site with Jetpack Beta pointing to this branch and enable sandbox access
* Click to activate Jetpack with one click
* Open the network tab in the console and filter by "recommendation"
* In the survey, click on "Grow my Audience" and submit the survey
* Confirm that the response of the request *does not* include any slug starting with `feature_`
* Now go to `/wp-admin/options-general.php?page=companion_settings` and update `JETPACK__SANDBOX_DOMAIN` to your sandbox domain.
* In your sandbox, checkout the branch `arcpatch-D168603`
* Go back to My Jetpack, click on the three dots at the top right and then "Redo" the recommendations. Click again on "Grow my audience" and watch the network requests
* Now, you should see `feature_related-posts` and `feature_newsletter` in the response
